src/components/tx/TxSimulation/utils.ts

```diff
@@ -129,19 +137,45 @@ const isSingleTransactionSimulation = (params: SimulationTxParams): params is Si
   return !Array.isArray(params.transactions)
 }
 
-export const getSimulationPayload = (params: SimulationTxParams): TenderlySimulatePayload => {
+/**
+ * @returns true for single MultiSig transactions if the provided signatures plus the current owner's signature (if missing)
+ * do not reach the safe's threshold.
+ */
+const isOverwriteThreshold = (params: SimulationTxParams) => {
+  if (!isSingleTransactionSimulation(params)) {
+    return false
+  }
+  const tx = params.transactions
+  const hasOwnerSig = tx.signatures.has(params.executionOwner)
+  const effectiveSigs = tx.signatures.size + (hasOwnerSig ? 0 : 1)
+  return params.safe.threshold > effectiveSigs
+}
+
+const getLatestBlockGasLimit = async (): Promise<number> => {
+  const web3 = getWeb3ReadOnly()
+  const latestBlock = await web3?.getBlock('latest')
+  return latestBlock?.gasLimit.toNumber() || 30_000_000
```

Comment:
We should normally put this in a constant, but in this case I would argue against hardcoding a value at all. We want the simulation to remain as 'true' as possible.

Reply:
The problem here is that `getBlock()` can return `undefined` this should never happen but for type safety I used this as a legit fallback.

Reply:
If there is no `gasLimit` and then `web3ReadOnly` is `undefined`, let's throw.

Reply:
okay